beacon_node/network/src/service.rs

```diff
@@ -465,13 +465,15 @@ fn spawn_service<T: BeaconChainTypes>(
                                 // We currently do not perform any action here.
                             },
                             BehaviourEvent::PeerDisconnected(peer_id) => {
-                            let _ = service
-                                .router_send
-                                .send(RouterMessage::PeerDisconnected(peer_id))
-                                .map_err(|_| {
-                                    debug!(service.log, "Failed to send peer disconnect to router");
-                                });
-                            },
+                                // Remove all subscriptions from peerdb for the disconnected peer.
+                                service.libp2p.swarm.peer_manager().remove_all_subscriptions(&peer_id);
```

Comment:
This may be safer to include it in the `inject_disconnected()` function. This way anytime they get disconnected (regardless of if the peer manager knows about it or not) we flush their subscriptions. 

The event here only gets called if the peer manager thinks we are connected. There are times when a new peer connects (but we have reached our peer limit so we dont inform the peer manager). In this case we could get subscriptions and the `BehaviourEvent::PeerDisconnected()` wont get called and we wont reset them when they disconnect.